add reorder atom in mol.py

v-shaoningli opened the pull request:
Add `reorder atom` function to achieve **atom-to-atom correspondence** in **PCQM4Mv2** when dealing with sdf.
The code is originally from [ViSNet](https://github.com/microsoft/ViSNet/blob/OGB-LSC%40NIPS2022/OGB_ViSNet/datasets/utils.py#L12).
We can test the order by
```
from rdkit import Chem
from ogb.utils import ReorderAtoms

suppl = Chem.SDMolSupplier('pcqm4m-v2-train.sdf')
mol = suppl[0]
mol = ReorderAtoms(mol)
atomic_number = []          
for atom in mol.GetAtoms():
    atomic_number.append(atom.GetAtomicNum())
print(atomic_number)
# >>> [6, 6, 6, 6, 8, 6, 6, 6, 6, 6, 8, 8, 6, 6, 6, 6, 7]
```

and 

```
from ogb.lsc import PCQM4Mv2Dataset
from ogb.utils import smiles2graph
from functools import partial

dataset = PCQM4Mv2Dataset(root = ROOT, smiles2graph=partial(smiles2graph, removeHs=True))
print(dataset[0].x[:, 0] + 1) # atomic number
# >>> [6, 6, 6, 6, 8, 6, 6, 6, 6, 6, 8, 8, 6, 6, 6, 6, 7]
```

weihua916 commented:
Thanks for the PR!

Just to confirm; this does not change the identity of the molecule, right? It just changes the ordering.

v-shaoningli commented:
Right. Since we need the `order` value in `ReorderAtoms` function to re-index the position in .sdf, we need to return this value.
For instance:

```
i = 0
mol = suppl[i]
mol, order = ReorderAtoms(mol)
order = torch.tensor(order).long()
N = mol.GetNumAtoms()
pos = suppl.GetItemText(i).split('\n')[4:4 + N]
pos = [[float(x) for x in line.split()[:3]] for line in pos]
pos = torch.tensor(pos)[order]
```

I would create a new PR to fix this.

weihua916 commented on ogb/utils/mol.py:
```diff
@@ -7,7 +7,7 @@
 def ReorderAtoms(mol):
```
Change the function name to reflect "how" you reorder the atom.

v-shaoningli replied:
Have changed the function name to be `CanonicalRankAtoms` in consistant to rdkit.

v-shaoningli replied:
Maybe we could convert SMILES and .sdf to graphs in `PygPCQM4Mv2Dataset` simultaneously.

weihua916 replied:
How about `ReorderCanonicalRankAtoms`?

And let's make sure backward compatibility not to disturb anyone who already calls this function.

v-shaoningli replied:
Sure. Have changed the function name and set the `reorder_atoms` as an option.

weihua916 replied:
thanks! LGTM.

v-shaoningli replied:
Would give a unified `PygPCQM4Mv2Dataset` later.

weihua916 replied:
Sounds good! Thanks. For the next PR, it would be nice for you to contribute to [example](https://github.com/snap-stanford/ogb/tree/master/examples/lsc/pcqm4m-v2). This way, we know and can test how to use these features.
